### PR TITLE
Add missing index.mapping.total_fields.limit setting to the target index

### PR DESCRIPTION
**Pull request by salvatore-campagna**
When the source index has a non-default value for the total number of fields in the mapping we need to copy that value and apply the same setting to the target index. This is necessary because later we apply the source index mappings to the target index. As a result, if the source index has a non-default limit on the number of fields in the mapping we need that same limit on the target index to avoid errors when applying the mappings.

### CODE REVIEW

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java [hunk 1]
```diff
@@ -558,6 +558,10 @@ private void createRollupIndex(
                 .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
                 .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "-1")
                 .put(IndexMetadata.INDEX_ROLLUP_STATUS.getKey(), IndexMetadata.RollupTaskStatus.STARTED)
+                .put(
+                    MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(),
+                    sourceIndexMetadata.getSettings().get(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
+                )
```
Comment (salvatore-campagna) [inline x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java:564]: I will update tests to check for this setting too.

**Review by nik9000 — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine test this please

**Comment by salvatore-campagna** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java [hunk 2]
```diff
@@ -558,6 +558,10 @@ private void createRollupIndex(
                 .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
                 .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "-1")
                 .put(IndexMetadata.INDEX_DOWNSAMPLE_STATUS.getKey(), IndexMetadata.DownsampleTaskStatus.STARTED)
+                .put(
+                    MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(),
+                    sourceIndexMetadata.getSettings().get(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
+                )
```
Comment (csoulios) [inline x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java:564]: Will this work if `MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING` is null at the source index? 
Do we know what happens in this case?

**Review by csoulios — APPROVED** [review]
LGTM!

I only left a comment if we should set the index setting when it's not set in source index.

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java (hunk 2 above)
Comment (salvatore-campagna) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java:564]: I can add a test for this maybe?

Comment (salvatore-campagna) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java:564]: Or better I can make the value random...

**Pull request merged**